docs/sequence/PISPDiscoveryApi.puml

```diff
@@ -0,0 +1,143 @@
+
+activate PISP_DM
+
+
+PISP_DM -> PISP_DM: ThirdpartyRequests.getAccounts()
+PISP_DM -> Switch: GET /accounts/username1234
+rnote right of PISP_DM #LightBlue
+GET /accounts/username1234
+FSIOP-Source: pispa
+FSIOP-Destination: dfspa
+end note
+
+activate Switch
+Switch --> PISP_DM: 202 Accepted
+deactivate PISP_DM
+
+Switch -> DFSP_TP_IN: GET /accounts/username1234
+activate DFSP_TP_IN
+DFSP_TP_IN --> Switch: 202 Accepted
+deactivate Switch
+
+DFSP_TP_IN -> DFSP_TP_IN: const model = await create()
+rnote right of DFSP_TP_IN: state: start
+DFSP_TP_IN -> DFSP_DM: model.retrieveAccounts()
+deactivate DFSP_TP_IN
+
+note over PISP, DFSP_DM
+  HAPPY_SCENARIO: DFSP returns list of accounts.
+end note
+
+activate DFSP_DM
+DFSP_DM -> DFSP_DM: DFSPBackendRequests.getUserAccounts('username1234')
+DFSP_DM -> DFSP_DM: Accounts returned.
+rnote right of DFSP_DM: state: accountsRetrieved
+
+rnote left of DFSP_DM #LightBlue
+NOTE: This deviates from the original design document.
+      Which has the response as a top-level array which
+      is a security and expandability risk.
+      Fixing that here.
+PUT /accounts/username1234
+FSIOP-Source: dfspa
+FSIOP-Destination: pispa
+{
+  "accounts": [
+    { accountNickname: "XXXXXXnt", id: "dfspa.username.1234", currency: "ZAR" },
+    { accountNickname: "SpeXXXXXXXXnt", id: "dfspa.username.5678", currency: "USD" }
+  ]
+}
+end note
+
+DFSP_DM -> Switch: PUT /accounts/username1234
+activate Switch
+Switch --> DFSP_DM: 200 OK
+deactivate DFSP_DM
+
+Switch --> PISP_TP_IN: PUT /accounts/username1234
+activate PISP_TP_IN
+PISP_TP_IN --> Switch: 200 OK
+deactivate Switch
+
+PISP_TP_IN -> PISP_DM: Accounts response recieved
+rnote right of PISP_DM: state: accountLookupSuccess
+deactivate PISP_TP_IN
+activate PISP_DM
+PISP_DM -> PISP_TP_OUT: return Accounts
+deactivate PISP_DM
+PISP_TP_OUT --> PISP: 200 OK Accounts
+
+
+note over PISP, DFSP_DM
+  ERROR_SCENARIO: Switch receives 7202 No accounts found for generic ID error.
+end note
+
+activate DFSP_DM
+DFSP_DM -> DFSP_DM: DFSPBackendRequests.getUserAccounts('username1234')
+DFSP_DM -> DFSP_DM: No accounts returned.
+DFSP_DM -> DFSP_DM: throw MojaloopFSPIOPError
+
+rnote left of DFSP_DM #LightCoral
+PUT /accounts/username1234/error
+FSIOP-Source: dfspa
+FSIOP-Destination: pispa
+{
+  errorInformation: {
+    errorCode: '7202',
+    errorDescription: 'No accounts found for generic ID'
+  }
+}
+end note
+
+DFSP_DM -> Switch: PUT /accounts/username1234/error
+activate Switch
+Switch --> DFSP_DM: 200 OK
+deactivate DFSP_DM
+
+Switch --> PISP_TP_IN: PUT /accounts/username1234/error
+activate PISP_TP_IN
+PISP_TP_IN --> Switch: 200 OK
+deactivate Switch
+
+PISP_TP_IN -> PISP_DM: MojaloopFSPIOPError response recieved
+deactivate PISP_TP_IN
+activate PISP_DM
+PISP_DM -> PISP_TP_OUT: return MojaloopFSPIOPError
+rnote right of PISP_DM: state: errored
+deactivate PISP_DM
+PISP_TP_OUT --> PISP: 500 Internal Server Error ErrorInformationObject
```

Comment:
For no accounts received, this would be a 404. 

500 should only really be for - "something went wrong in processing this request"

Reply:
thanks